lib/with_locking.rb

```diff
@@ -25,12 +25,23 @@ def self.run!(options = {}, &block)
     raise "locked process still running" unless self.run(options, &block)
   end
 
+  def self.remove_staled_file(options)
+    return unless File.exists?(pidfile(options)) && options.include?(:stale_seconds)
+    File.delete(pidfile(options)) if pid_staled(options)
+  end
+
+  def self.pid_staled(options)
```

Comment:
I really want specs, because I'm not sure this logic is correct.

Also, to make the code easier to understand, I'd prefer to see something like this:

```ruby
File.ctime(pidfile(options)).utc.to_i <Time.now.utc.to_i - options.fetch(:stale_seconds)
```

This says "if the file's create time is older than 4 hours ago, then it's really stale".